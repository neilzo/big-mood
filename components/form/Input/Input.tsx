import React from 'react';
import { StyleSheet, View, TextInput, Text } from 'react-native';
import colorVariables from '../../colorVariables';

interface Props {
  onChange: (text: string) => void;
  label: string;
  value: string;
}
const Input = ({ onChange, label, value }: Props) => (
  <View style={styles.container}>
    {label && <Text>{label}</Text>}
    <TextInput onChangeText={onChange} value={value} />
  </View>
);

const styles = StyleSheet.create({
  container: {
    alignSelf: 'stretch',
    marginTop: 15
  },
  label: {},
  input: {
    backgroundColor: colorVariables.white,
    borderWidth: 1,
    borderRadius: colorVariables.borderRadius,
    borderColor: colorVariables.borderColor,
    alignSelf: 'stretch',
    height: 50,
    padding: 15
  }
});

export default Input;
