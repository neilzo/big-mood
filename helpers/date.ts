import moment from 'moment';

const getPrettyDate = (date: Date) => moment(date).format('MMM Do YYYY');

const getPrettyTime = (date: Date) => moment(date).format('h:mma');

export default { getPrettyDate, getPrettyTime };
