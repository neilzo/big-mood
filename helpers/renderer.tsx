import 'react-native';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

export const createComponent = (component: any) => {
  return renderer.create(component);
};

// export const createComponentWithRedux = (children: any, props = {}) => {
//   return renderer.create(<Provider {...props}>{children}</Provider>);
// };
