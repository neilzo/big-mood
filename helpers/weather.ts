interface WeatherResults {
  weather: Array<{ main: string; icon: string }>;
  main: {
    temp: number;
  };
}

const kelvinToFarenheit = (kelvin: number) => {
  return Math.round(kelvin * (9 / 5) - 459.76);
};

const computeOverallWeather = (weatherResults: WeatherResults) => ({
  icon: weatherResults.weather[0].icon,
  description: weatherResults.weather[0].main,
  temperature: kelvinToFarenheit(weatherResults.main.temp),
});

export { kelvinToFarenheit, computeOverallWeather };
