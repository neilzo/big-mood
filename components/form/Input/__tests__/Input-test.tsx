import React from 'react';
import Input from '../Input';
import { createComponent } from '../../../../helpers/renderer'; // todo resolve imports to avid that ugly thang

const mockProps = {
  onChange: () => {},
  value: 'someVal',
};

describe('Input', () => {
  it('renders correctly', () => {
    expect(
      createComponent(<Input {...mockProps} />).toJSON()
    ).toMatchSnapshot();
  });

  it('renders with a label', () => {
    expect(
      createComponent(<Input {...mockProps} label="Suh Dude" />).toJSON()
    ).toMatchSnapshot();
  });
});
