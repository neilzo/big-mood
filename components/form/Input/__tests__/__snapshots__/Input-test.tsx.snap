// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "marginTop": 15,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#fff",
        "borderColor": "#ccc",
        "borderRadius": 3,
        "borderWidth": 1,
        "height": 50,
        "padding": 15,
      }
    }
    underlineColorAndroid="transparent"
    value="someVal"
  />
</View>
`;

exports[`Input renders with a label 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "marginTop": 15,
    }
  }
>
  <Text
    style={
      Object {
        "marginBottom": 10,
      }
    }
  >
    Suh Dude
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#fff",
        "borderColor": "#ccc",
        "borderRadius": 3,
        "borderWidth": 1,
        "height": 50,
        "padding": 15,
      }
    }
    underlineColorAndroid="transparent"
    value="someVal"
  />
</View>
`;
